Azure/azure-service-operator
# allow redis cache to be deployed into a vnet

closes #797 

**What this PR does / why we need it**:
exposes the fields necessary to put redis in a vnet

To test:

- create resource group
- create virtual network in rg
- set subnet id and an ip from within the vnet subnet range
- deployment should succeed and the portal should show that it is in a vnet
